packages/react-components/src/components/VideoGallery/useVideoStreamLifecycleMaintainer.ts

```diff
@@ -6,22 +6,45 @@ import { VideoStreamOptions, CreateVideoStreamViewResult, ScalingMode } from '..
 import { Cancellable, useCancellableTask } from '../utils/useCancellableTask';
 
 /**
- * Helper hook to maintain the video stream lifecycle. This calls onCreateStreamView and onDisposeStreamView
- * appropriately based on react lifecycle events and prop changes.
- * This also handles calls to view.update* appropriately such as view.updateScalingMode().
- *
+ * Props needed for {@link useVideoStreamLifecycleMaintainer}
  * @private
  */
-export const useVideoStreamLifecycleMaintainer = (props: {
+export interface VideoStreamLifecycleMaintainerProps {
   isStreamAvailable?: boolean;
   renderElementExists?: boolean;
   isMirrored?: boolean;
   scalingMode?: ScalingMode;
-  onCreateStreamView?: (options?: VideoStreamOptions) => Promise<void | CreateVideoStreamViewResult>;
-  onDisposeStreamView?: () => void;
-}): void => {
-  const { onCreateStreamView, onDisposeStreamView, isStreamAvailable, renderElementExists, isMirrored, scalingMode } =
-    props;
+  remoteParticipantId?: string;
```

Comment:
Please add a comment here noting that local stream is created if this is undefined.

Reply:
Refactor no longer requires this